File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        int minNum = Integer.MAX_VALUE;
        for(int num : nums) {
            minNum = Math.min(minNum, getDigitSum(num));
        }
        return minNum;
    }
    int getDigitSum(int num) {
        int sum = 0;
        for(char c : String.valueOf(num).toCharArray()) {
            sum += (c-'0');
        }
        return sum;
    }
}

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minElement = function(nums) {
    let minNum = Number.MAX_SAFE_INTEGER;

    for(let num of nums) {
        minNum = Math.min(minNum, getDigitSum(num));
    }
    return minNum;
};

var getDigitSum = function(num) {
    let sum = 0;
    for(let c of num.toString()) {
        sum += Number(c);
    }
    return sum;
}